hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/MetricAggregationMapConverter.java

```diff
@@ -57,39 +57,51 @@
             this.aggregationConverter.convert(resultMap.get(aggregationRequest.alias()).getValue()),
             this.baselineAggregationConverter.convert(
                 Optional.ofNullable(baselineMap.get(aggregationRequest.alias()))),
-            (metricAggregation, baselinedMetricAggregation) ->
-                new BaselinedMetricAggregation() {
-                  @Override
-                  public Double value() {
-                    return metricAggregation.value();
-                  }
-
-                  @Override
-                  public MetricBaselineAggregation baseline() {
-                    return new MetricBaselineAggregation() {
-
-                      @Override
-                      public Double value() {
-                        return baselinedMetricAggregation.value();
-                      }
-
-                      @Override
-                      public Double lowerBound() {
-                        return baselinedMetricAggregation.lowerBound();
-                      }
-
-                      @Override
-                      public Double upperBound() {
-                        return baselinedMetricAggregation.upperBound();
-                      }
-
-                      @Override
-                      public Health health() {
-                        return baselinedMetricAggregation.health();
-                      }
-                    };
-                  }
-                })
+            BaselinedMetricAggregationImpl::new)
         .map(agg -> Map.entry(MetricLookupMapKey.fromAggregationRequest(aggregationRequest), agg));
   }
+
+  private static class MetricBaselineAggregationImpl implements MetricBaselineAggregation {
+
+    private final MetricBaselineAggregation metricBaselineAggregation;
+
+    public MetricBaselineAggregationImpl(MetricBaselineAggregation metricBaselineAggregation) {
```

Comment:
What is the purpose of this class? It looks like its only constructor is a clone constructor

Reply:
Removed, not required